src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs

```diff
@@ -26,9 +27,10 @@ internal class HostListenerFactory : IListenerFactory
         private readonly ILoggerFactory _loggerFactory;
         private readonly ILogger _logger;
         private readonly bool _allowPartialHostStartup;
+        private readonly IScaleMonitorManager _monitorManager;
 
         public HostListenerFactory(IEnumerable<IFunctionDefinition> functionDefinitions, SingletonManager singletonManager, IJobActivator activator,
-            INameResolver nameResolver, ILoggerFactory loggerFactory, bool allowPartialHostStartup = false)
+            INameResolver nameResolver, ILoggerFactory loggerFactory, IScaleMonitorManager monitorManager, bool allowPartialHostStartup = false)
```

Comment:
I introduced this IScaleMonitorManager as a registry for monitor instances, to solve the problem that these instances are discovered/registered at *runtime* below. This way, we only register the monitor for a listener if it isn't a disabled function. Open to better ways of doing this. I didn't think I should be taking and dynamically modifying the DI services collection.